### PR TITLE
GAWB-735: Show dialog when anything returns 503

### PR DESCRIPTION
**Pull request by MattPutnam**
Right now there isn't any useful information in the 503, so the message is sorta generic.  The load balancing work will address this, at which point we can make the message nicer.

Note: I tested this locally by changing "503" to other things.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
I would love to see, as a general rule, more comments :)

`(fn [_ _ _ ns]` is reminiscent of:

```
@P=split//,".URRUU\c8R";@d=split//,"\nrekcah xinU / lreP rehtona tsuJ";sub p{
@p{"r$p","u$p"}=(P,P);pipe"r$p","u$p";++$p;($q*=2)+=$f=!fork;map{$P=$P[$f^ord
($p{$_})&6];$p{$_}=/ ^$P/ix?$P:close$_}keys%p}p;p;p;p;p;map{$p{$_}=~/^[P.]/&&
close$_}%p;wait until$?;map{/^r/&&<$_>}%p;$_=$d[$q];sleep rand(2)if/\S/;print
```

other than that, 👍

**Comment by rushtong** [comment]
+1 on comments.

File: src/cljs/org/broadinstitute/firecloud_ui/common/dialog.cljs [hunk 1]
```diff
@@ -16,26 +16,26 @@
       :cycle-focus? false})
    :render
    (fn [{:keys [props state]}]
-     (let [content (:content props)
-           anchored? (not (nil? (:get-anchor-dom-node props)))]
+     (let [{:keys [content width blocking? dismiss-self get-anchor-dom-node]} props
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/dialog.cljs:19]: Really loving this pattern.

**Comment by dmohs** [comment]
The underscore is a common idiom (same as Scala) that just means, "I don't care about these parameters." I've seen this usage of `add-watch` in Clojure's docs:

``` clojure
(add-watch my-atom :some-key (fn [k r os ns] ...))
```

so the usage in this PR makes sense (since we don't care about the first three parameters).

However, I disagree with Clojure's docs in terms of variable naming here. The names `os` and `ns` are not descriptive enough and `ns` is a built-in for declaring a namespace, making this even worse. In our calls of `add-watch`, I'd suggest we use a more descriptive pattern:

``` clojure
(fn [_ _ _ new-value]
  (when new-value
    (swap! state assoc :show-server-down-message? true))))
```

No need to hold up this PR, but it'd be cool to see a small refactoring PR that fixes both calls to `add-watch` in our codebase. With that said, 👍

**Comment by davidangb** [comment]
I'm not at all debating the proper use of syntax (or underscores) here ... I also ask for comments within Scala code!

**Comment by davidangb** [comment]
👍

**Pull request merged**